documentation/scaling_puppet_server.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: default
+title: "Scaling Puppet Server with compile masters"
+canonical: "/puppetserver/latest/scaling_puppet_server.html"
+---
+
+To scale Puppet Server for many thousands of nodes, you'll need to add Puppet masters dedicated to catalog compilation. These Servers are known as **compile masters**, and are simply additional load-balanced Puppet Servers that receive catalog requests from agents and synchronize the results with each other.
+
+## Planning your load-balancing strategy
+
+The rest of your configuration depends on how you plan on distributing the agent load. Determine what your deployment will look like before you add any compile masters, but **implement load balancing as the last step** only after you have the infrastructure in place to support it.
+
```

Comment:
This is aimed at open-source Server users. If we're worried about PE users landing here, we should point back to relevant PE docs, but if there's relevant content in the PE docs we should just fold the relevant parts into here. (And vice-versa.)